packages/react-devtools-shared/src/__tests__/HooksTree-test.js

```diff
@@ -0,0 +1,92 @@
+describe('HooksTree', () => {
+  let React;
+  let ReactDOM;
+  let bridge: FrontendBridge;
+  let store: Store;
+  let utils;
+  let HooksTree;
+
+  let BridgeContext;
+  let InspectedElementContextController;
+  let StoreContext;
+  let TreeContextController;
+
+  beforeEach(() => {
+    utils = require('./utils');
+    utils.beforeEachProfiling();
+
+    bridge = global.bridge;
+    store = global.store;
+    store.collapseNodesByDefault = false;
+
+    React = require('react');
+    ReactDOM = require('react-dom');
+
+    BridgeContext = require('react-devtools-shared/src/devtools/views/context')
+      .BridgeContext;
+    InspectedElementContextController = require('react-devtools-shared/src/devtools/views/Components/InspectedElementContext')
+      .InspectedElementContextController;
+    StoreContext = require('react-devtools-shared/src/devtools/views/context')
+      .StoreContext;
+    TreeContextController = require('react-devtools-shared/src/devtools/views/Components/TreeContext')
+      .TreeContextController;
+
+    HooksTree = require('react-devtools-shared/src/devtools/views/Components/HooksTree')
+      .HooksTreeView;
+  });
+
+  const Contexts = ({
+    children,
+    defaultSelectedElementID = null,
+    defaultSelectedElementIndex = null,
+  }) => (
+    <BridgeContext.Provider value={bridge}>
+      <StoreContext.Provider value={store}>
+        <TreeContextController
+          defaultSelectedElementID={defaultSelectedElementID}
+          defaultSelectedElementIndex={defaultSelectedElementIndex}>
+          <InspectedElementContextController>
+            {children}
+          </InspectedElementContextController>
+        </TreeContextController>
+      </StoreContext.Provider>
+    </BridgeContext.Provider>
+  );
+
+  it('shows the complex value of custom hooks with sub-hooks', () => {
+    // props from `InspectedElementContext display complex values of useDebugValue: DisplayedComplexValue 1` snapshot
+
+    const container = document.createElement('div');
+    ReactDOM.render(
+      <Contexts>
+        <HooksTree
+          hooks={[
```

Comment:
What is this test meant to be testing? I'm not sure I see the value it adds.

Hard-coding the hooks metadata doesn't seem like it's a very comprehensive test. I think the existing pattern of testing via the contexts is a better end-to-end coverage. Less brittle. :)

Reply:
That was my suggestion about. Directly injecting the tree.hooks into the prop. I couldn't find an existing e2e test for the react-devtools view.

Reply:
I don't think there's a lot of value in testing the view components.  DevTools tests the underlying business logic on the backend (and frontend via the contexts).

Let's remove this test :)